File: rust/examples/basic.rs
use ratatouille::{Format, Logger, LoggerConfig, SourceIdentity};

fn main() {
    let mut config = LoggerConfig::default();
    config.filter = Some("api*,-api:noise".to_owned());
    config.format = Format::Ndjson;
    config.source = SourceIdentity {
        app: Some("example".to_owned()),
        r#where: Some("rust".to_owned()),
        instance: Some("local".to_owned()),
    };

    let mut logger = Logger::new(config);
    let _ = logger.log("api", "hello from Rust");
    let _ = logger.log("api", "user=alice req=42");
    let _ = logger.log("api:noise", "this should be filtered");

    let stats = logger.stats();
    eprintln!(
        "emitted={} filtered={} dropped={} topics={}",
        stats.emitted, stats.filtered, stats.dropped, stats.known_topics
    );
}

File: c/examples/basic.c
/* SPDX-FileCopyrightText: 2026 Alexander R. Croft */
/* SPDX-License-Identifier: MIT */

#include "ratatouille.h"

#include <stdio.h>

int main(void) {
    rat_config_t cfg = {0};
    rat_logger_t *log;
    rat_stats_t stats;

    cfg.filter = "api*,-api:noise";
    cfg.format = RAT_FORMAT_NDJSON;
    cfg.source.app = "example";
    cfg.source.where = "c";
    cfg.source.instance = "local";
    cfg.max_topics = 32;

    log = rat_logger_create(&cfg);
    if (!log) {
        fprintf(stderr, "failed to create ratatouille logger\n");
        return 1;
    }

    rat_log(log, "api", "hello from C");
    rat_logf(log, "api", "user=%s req=%d", "alice", 42);
    rat_log(log, "api:noise", "this should be filtered");

    stats = rat_logger_stats(log);
    fprintf(
        stderr,
        "emitted=%llu filtered=%llu dropped=%llu topics=%llu\n",
        (unsigned long long)stats.emitted,
        (unsigned long long)stats.filtered,
        (unsigned long long)stats.dropped,
        (unsigned long long)stats.known_topics
    );

    rat_logger_destroy(log);
    return 0;
}
